gum/arch-arm64/gumarm64writer.c

```diff
@@ -982,3 +999,94 @@ gum_arm64_writer_describe_reg (GumArm64Writer * self,
   }
   ri->index = ri->meta - GUM_MREG_R0;
 }
+
+void
+gum_arm64_writer_put_push_all_registers (GumArm64Writer * self)
+{
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X0, ARM64_REG_X1);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X2, ARM64_REG_X3);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X4, ARM64_REG_X5);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X6, ARM64_REG_X7);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X8, ARM64_REG_X9);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X10, ARM64_REG_X11);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X12, ARM64_REG_X13);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X14, ARM64_REG_X15);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X16, ARM64_REG_X17);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X18, ARM64_REG_X19);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X20, ARM64_REG_X21);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X22, ARM64_REG_X23);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X24, ARM64_REG_X25);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X26, ARM64_REG_X27);
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X28, ARM64_REG_X29);
+  gum_arm64_writer_put_instruction (self, 0xD53B420F); /* MRS X15, NZCV */
+  gum_arm64_writer_put_push_reg_reg (self, ARM64_REG_X30, ARM64_REG_X15);
+}
+
+void
+gum_arm64_writer_put_push_all_Q_registers (GumArm64Writer * self)
+{
+  guint32 instructions[] = {
+    0xADBF07E0,
+    0xADBF0FE2,
+    0xADBF17E4,
+    0xADBF1FE6,
+    0xADBF27E8,
+    0xADBF2FEA,
+    0xADBF37EC,
+    0xADBF3FEE,
+    0xADBF47F0,
+    0xADBF4FF2,
+    0xADBF57F4,
+    0xADBF5FF6,
+    0xADBF67F8,
+    0xADBF6FFA,
+    0xADBF77FC,
+    0xADBF7FFE
+  };
+  gum_arm64_writer_put_bytes (self, instructions, sizeof(instructions));
+}
+
+void
+gum_arm64_writer_put_pop_all_registers (GumArm64Writer * self)
+{
+  gum_arm64_writer_put_pop_reg_reg (self, ARM64_REG_X30, ARM64_REG_X15);
+  gum_arm64_writer_put_instruction (self, 0xD51B420F); /* msr NZCV, x15 */
```

Comment:
`gum_arm64_writer_put_push_all_registers`  and `gum_arm64_writer_put_pop_all_registers` actually pushes and pops also the NZCV register. Comments fixed.